src/components/CharacterAssignment.jsx

```diff
@@ -0,0 +1,229 @@
+import React, { useEffect, useState } from 'react';
+import { DragDropContext } from 'react-beautiful-dnd';
+import ActorColumn from './ActorColumn.jsx'
+import UnassignedColumn from './UnassignedColumn';
+
+// Entire Drag and Drop context for assigning characters to actors
+const CharacterAssignment = ({actors, currentScript}) => {
+  const [columnOrder, setColumnOrder] = useState([])
+  const [columns, setColumns] = useState({
+    'unassignedCharacters': {
+      id:'unassignedCharacters',
+        title: 'Unassigned Characters',
+        characterIds: [],
+        lineCount: 0
+    }});
+  const [characters, setCharacters] = useState()
+
+  // Create the actor columns initially and edit columns when actors are added or removed
+  useEffect(() => {createOrUpdateActorColumns(actors)}, [actors]);
```

Comment:
How come these are useEffects? I thought using react router would mean these should be loaders? Is that incorrect?

Reply:
I did look into this, I believe for us to take advantage of the loaders we would have to make the CharacterAssignment component an outlet (https://reactrouter.com/en/main/components/outlet) of the Home page. I tried implementing this and ran into issues with the component wanting to render before it had the data it needed from the loader. I'm unsure if this is because as an outlet, its parent component already exists and has its own loader, if loaders aren't entirely compatible with Outlet components, or if I just don't understand enough about how everything is interacting to implement it properly.

Since we will be splitting this component out in the future, I think we should leave it as is and once it's on its own page it will be easier to add a loader.